### DISCUSSION TITLE
When drawing a set of normal data, mark some of the abnormal data points or give them different colors.

### DISCUSSION THREAD

**Discussion by ningwana** [pyqtgraph/pyqtgraph#2635] [Q&A] [answered]
Hello, I just used this library, and I think it's great, but I'm having some problems using it.
When I plot a set of normal data, I want to mark the abnormal data points or give them a different color.

```python
import pyqtgraph as pg
import numpy as np

w = pg.GraphicsLayoutWidget()
w.show()

x = np.arange(10)  # [0 1 2 3 4 5 6 7 8 9]
y = np.arange(10) % 3  # [0 1 2 0 1 2 0 1 2 0]

# =====================================
# [0,0],[1,1],[2,2],[3,0],[4,1],[5,2],[6,0],[7,1],[8,2],[9,0]
# Now I know that some of the data is abnormal data(eg: [3,0],[4,1],[6,0]), 
# but how do I plot it? Like giving different colors.
# =====================================

plt = w.addPlot(row=0, col=0)
plt.plot(x, y, symbol='o', pen={'color': 0.8, 'width': 2})

if __name__ == '__main__':
    pg.exec()
```

**Comment by pijyoi** [comment, marked as answer]
```python
import pyqtgraph as pg
import numpy as np

pg.mkQApp()
pw = pg.PlotWidget()
pw.show()

x = np.arange(10)  # [0 1 2 3 4 5 6 7 8 9]
y = np.arange(10) % 3  # [0 1 2 0 1 2 0 1 2 0]

mask = [True, True, True, False, False, True, False, True, True, True]
symbol = ['o' if x else 'x' for x in mask]
g = pg.mkBrush('g')
r = pg.mkBrush('r')
symbolBrush = [g if x else r for x in mask]

pw.plot(x, y, symbol=symbol, symbolBrush=symbolBrush)

pg.exec()
```

**Reply by ningwana** [reply]
Thank you very much for your response, it helped me solve the problem.

Thanks again

**Reply by ningwana** [reply]
@pijyoi 

When I have a lot of data, such as hundreds of thousands of data, the drawing will be very stuck

Here's what I do, I compare the ids of the two pieces of data（abnormal data and normal data）, and then I define the symbol and symbolbrush for each point, and then I go to plot...

And then it got stuck.

**Reply by pijyoi** [reply]
```python
import pyqtgraph as pg
import numpy as np

pg.mkQApp()
pw = pg.PlotWidget()
pw.show()

ttt = np.linspace(0, 1, 100_000)
data = np.sin(2*np.pi*100*ttt) * (2 + np.sin(2*np.pi*ttt))
mask = np.abs(data) < 2

items = []
items.append(pg.PlotDataItem(ttt, data))
items.append(pg.PlotDataItem(ttt[~mask], data[~mask], symbol='x', symbolBrush='r'))
items.append(pg.PlotDataItem(ttt[mask], data[mask], symbol='o', symbolBrush='g'))
for item in items:
    pw.addItem(item)
    item.setDownsampling(auto=True)
    item.setClipToView(True)

pg.exec()
```

**Reply by ningwana** [reply]
@pijyoi 
Wow, it's silky, and I'm glad you could answer it again.
That really helped me a lot and gave me a better understanding of the library(pyqtgraph).

**Reply by ningwana** [reply]
@pijyoi  

I have a button that I click to draw. When I finish drawing for the first time, I use the mouse wheel to zoom in or out the pattern, and click again, the axis does not automatic update range, it is still the axis range that I zoom in the graph for the first time, which causes the display to be abnormal, and then when I click view all. It can be displayed properly.

**Reply by ningwana** [reply]
I'm glad I found a solution, just set the enableAutoRange to do it.
